requirements.txt

```diff
@@ -1,2 +1,5 @@
 # generated from manifests external_dependencies
+dataclasses
+odoorpc
+openupgradelib
```

Comment:
Don't you need to do a git install from master for openupgradelib? AFAIK the pypi version is always bad...

Reply:
I confirm it's a bear trap to install openupgrade lib from pypi

Reply:
Well. Not sure to understand. When executing a migration with openupgrade project, yes we have to have the up to date library, because some function are oftently added / refactored / etc...

But for that module (``upgrade_analysis``) it is using only a function ``table_exists`` [here](https://github.com/OCA/server-tools/pull/1941/files#diff-cc8b10fd780fc230005882ef34b0ad0f32d7e66fe3ee501f299f3a65e3b25005R7) and that function is quite simple & stable.
https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade_tools.py#L30

But, if you prefer, I can replace ``openupgradelib`` by ``git+https://github.com/OCA/openupgradelib.git@master``

Reply:
It's better to always put the latest one, as we can even patch that method or improve this module using other methods of `openupgradelib` when needed.

Reply:
done.

Reply:
Hi. I have a problem with that change. It is making fail the pre-commit process : 

```
Generate requirements.txt for an addons directory........................Failed
- hook id: setuptools-odoo-get-requirements
- files were modified by this hook
```
Could you say me what to do ? 

thanks !

Reply:
ah... 🤦🏼‍♂️ 

Could you please try just removing the openupgradelib line from here? I guess that dependency should be present in CIs always... (but I'm not sure)

Reply:
Well, I can not remove this line, because pre-commit readd this line ;-)

``Generate requirements.txt for an addons directory``

I so commited again what pre-commit suggests. (I mean ``openupgradelib`` in ``requirements.txt``)

I propose to merge this PR as it. if someone see a better implementation, feel free to improve in another PR.